### DISCUSSION TITLE
facing HTTP 500 Internal Server Error Failed to start the session because headers have already been sent by "" at line 0 error while sending response from controller to ext js

### DISCUSSION THREAD

**Discussion by Thejaswini-Rao-U** [pimcore/pimcore#17394] [General]
### Pimcore version

10.6.9

### Steps to reproduce

When i send response to js , gives invlaid json error, and in the server i receive below error

HTTP 500 Internal Server Error Failed to start the session because headers have already been sent by "" at line 0 error while sending response from controller to ext js

![headererror](https://github.com/user-attachments/assets/47f628b6-dee6-48d0-b5ab-068f31664fce)

In js invalid json error
![invalidJson](https://github.com/user-attachments/assets/e687c257-5471-4de1-8a9b-6c266ce9143e)

Can you please give the solution to solve this error

### Actual Behavior

not working

### Expected Behavior

should work

**Comment by kingjia90** [comment]
Any `echo` or `print` statement used in the code that may print an empty string?
This doesn't seem coming/caused by Pimcore itself but from custom code. 
Switching to discussion area, also because 10.6 is EOL.

The best way to avoid it is to be sure there are no print nor echo statements, and you are only returning
via `$this->adminJson()` eg,
https://github.com/pimcore/admin-ui-classic-bundle/blob/0e14ad32b3b97a68c4feddc664589f292fa36194/src/Controller/Admin/Asset/AssetController.php#L107

**Reply by Thejaswini-Rao-U** [reply]
@kingjia90, Thanks for the quick response.

I have rechecked it, there is no echo or print statement in the code, also im returning response in $this->json($response); format. when i print the data just before return it could give the all data needed, as soon as i return and check it gives me the error, not sure how to solve it?
